### PR TITLE
Add column.slice_rows and column.get_rows_by_mask

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Here's another couple of dataframe methods missing for column

### CODE REVIEW

**Review by kkraus14 — APPROVED** [review]

File: spec/API_specification/dataframe_api/column_object.py [hunk 1]
```diff
@@ -92,6 +92,46 @@ def get_rows(self: Column[DType], indices: Column[Any]) -> Column[DType]:
...
+        mask : Column[bool]
+
+        Returns
+        -------
+        Column
+
+        Notes
+        -----
+        Some participants preferred a weaker type Arraylike[bool] for mask,
+        where 'Arraylike' denotes an object adhering to the Array API standard.
```
Comment (rgommers) [inline spec/API_specification/dataframe_api/column_object.py:130]: I think it's fine to leave this note here for now, useful to capture the differing opinions while moving forward. We can remove it once we polish things for a final release.

File: spec/API_specification/dataframe_api/column_object.py [hunk 2]
```diff
@@ -92,6 +92,46 @@ def get_rows(self: Column[DType], indices: Column[Any]) -> Column[DType]:
         """
         ...
 
+
+    def slice_rows(
+        self: Column[DType], start: int | None, stop: int | None, step: int | None
```
Comment (rgommers) [inline spec/API_specification/dataframe_api/column_object.py:97]: Allowing both positional and keyword usage seems good to me here.

**Review by rgommers — APPROVED** [review]
LGTM, thanks Marco.

File: spec/API_specification/dataframe_api/column_object.py (hunk 1 above)
Comment (MarcoGorelli) [reply spec/API_specification/dataframe_api/column_object.py:130]: thanks - I just copied it from the dataframe counterpart

**Pull request merged**